Source/Core/VideoCommon/VertexLoaderARM64.cpp

```diff
@@ -0,0 +1,481 @@
+// Copyright 2013 Dolphin Emulator Project
+// Licensed under GPLv2
+// Refer to the license.txt file included.
+
+#include "Core/PowerPC/JitArm64/Jit.h"
+#include "VideoCommon/VertexLoaderARM64.h"
+
+using namespace Arm64Gen;
+
+ARM64Reg src_reg = X0;
+ARM64Reg dst_reg = X1;
+ARM64Reg count_reg = W2;
+ARM64Reg skipped_reg = W17;
+ARM64Reg scratch1_reg = W16;
+ARM64Reg scratch2_reg = W15;
+ARM64Reg scratch3_reg = W14;
+ARM64Reg scratch4_reg = W13;
+ARM64Reg saved_count = W12;
+
+VertexLoaderARM64::VertexLoaderARM64(const TVtxDesc& vtx_desc, const VAT& vtx_att)
+	: VertexLoaderBase(vtx_desc, vtx_att), m_float_emit(this)
+{
+	if (!IsInitialized())
+		return;
+
+	AllocCodeSpace(4096);
+	ClearCodeSpace();
+	GenerateVertexLoader();
+	WriteProtect();
+}
+
+void VertexLoaderARM64::GetVertexAddr(int array, u64 attribute, ARM64Reg reg)
+{
+	ADD(reg, src_reg, m_src_ofs);
+	if (attribute & MASK_INDEXED)
+	{
+		if (attribute == INDEX8)
+		{
+			LDRB(INDEX_UNSIGNED, scratch1_reg, reg, 0);
+			m_src_ofs += 1;
+		}
+		else
+		{
+			LDRH(INDEX_UNSIGNED, scratch1_reg, reg, 0);
+			m_src_ofs += 2;
+			REV16(scratch1_reg, scratch1_reg);
+		}
+		if (array == ARRAY_POSITION)
+		{
+			if (attribute == INDEX8)
+			{
+				CMP(scratch1_reg, 0xFF);
+			}
+			else
+			{
+				MOVI2R(scratch2_reg, 0xFFFF);
+				CMP(scratch1_reg, scratch2_reg);
```

Comment:
This 0xffff can't be an immediate? I thought the immediate encoding was flexible enough for a simple bit pattern like that (-1)